addons/mass_mailing/migrations/9.0.2.0/pre-migration.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# © 2016 Tecnativa - Vicent Cubells <vicent.cubells@tecnativa.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+column_copies = {
+    'mail_mass_mailing': [
+        ('state', None, None),
+    ],
+}
+
+column_renames = {
+    'mail_mass_mailing_campaign': [
+        ('category_ids', None),
```

Comment:
As this is a many2many, you can't rename a column, but rename the table to avoid the removal.